src/talk/renderer/Viewer/Viewer.js

```diff
@@ -0,0 +1,83 @@
+/*
+ * @copyright Copyright (c) 2024 Grigorii Shartsev <me@shgk.me>
+ *
+ * @author Grigorii Shartsev <me@shgk.me>
+ *
+ * @license AGPL-3.0-or-later
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import Vue from 'vue'
+import ViewerApp from './ViewerApp.vue'
+import ViewerHandlerImages from './ViewerHandlerImages.vue'
+import ViewerHandlerVideos from './ViewerHandlerVideos.vue'
+
+export const Viewer = {
+	availableHandlers: [{
+		id: 'images',
+		group: 'media',
+		mimes: [
+			'image/apng',
+			'image/bmp',
+			'image/gif',
+			'image/jpeg',
+			'image/png',
+			'image/svg+xml',
+			'image/webp',
+			'image/x-icon',
+			'image/x-xbitmap',
+		],
+		component: ViewerHandlerImages,
+	}, {
+		id: 'videos',
+		group: 'media',
+		mimes: [
+			'video/mpeg',
+			'video/ogg',
+			'video/webm',
+			'video/mp4',
+			'video/x-m4v',
+			'video/x-flv',
+			'video/quicktime',
+			'video/x-matroska',
+		],
+		component: ViewerHandlerVideos,
+	}],
+
+	async open(...args) {
+		if (!Viewer._viewer) {
+			await Viewer.mount()
+		}
+
+		Viewer._viewer.open(...args)
+	},
+
+	close() {
+		if (!Viewer._viewer) {
+			return
+		}
+		Viewer._viewer.close()
+	},
+
+	mount() {
+		const container = document.createElement('div')
+		container.id = 'viewer'
+		document.body.appendChild(container)
```

Comment:
Check for fullscreen mode

Reply:
Fixed in a different way to support original reparentViewer from Talk:
- Mount viewer when the app is initialized, so one nextTick is enought
- add `id="viewer"`